### PR TITLE
[Pytorch][Graviton][Build][Sagemaker] PT 1.12 Graviton SageMaker Release

### PR DESCRIPTION
**Pull request by xncqr**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
This is adding the sagemaker image to the Pytorch 1.12 release

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by xncqr** [comment]
This PR is on hold baring some issues with SageMaker

**Review by kenny-ezirim — APPROVED** [review]

File: pytorch/buildspec-graviton.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
 region: &REGION <set-$REGION-in-environment>
 framework: &FRAMEWORK pytorch
 version: &VERSION 1.12.1
-os_version: &OS_VERSION ubuntu20.04
+
```
Comment (saimidu) [inline pytorch/buildspec-graviton.yml:5]: nit: Please remove this blank line.

File: pytorch/buildspec-graviton.yml [hunk 2]
```diff
@@ -31,9 +34,22 @@ images:
...
     image_size_baseline: 10000
     device_type: &DEVICE_TYPE cpu
+    os_version: &OS_VERSION ubuntu20.04
     python_version: &DOCKER_PYTHON_VERSION py3
     tag_python_version: &TAG_PYTHON_VERSION py38
     tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *OS_VERSION , "-ec2"]
-    docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /Dockerfile.graviton.ec2., *DEVICE_TYPE ]
+    docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /Dockerfile.graviton., *DEVICE_TYPE ]
     context:
       <<: *INFERENCE_CONTEXT
```
Comment (saimidu) [inline pytorch/buildspec-graviton.yml:43]: Is this spec missing the `target` input, to tell it that the build target is only `ec2`?

File: pytorch/buildspec-graviton.yml [hunk 3]
```diff
@@ -31,9 +34,22 @@ images:
...
+    <<: *INFERENCE_REPOSITORY
+    build: &PYTORCH_CPU_INFERENCE_PY3 false
+    image_size_baseline: 10000
+    device_type: &DEVICE_TYPE cpu
+    os_version: &OS_VERSION ubuntu20.04
+    python_version: &DOCKER_PYTHON_VERSION py3
+    tag_python_version: &TAG_PYTHON_VERSION py38
+    tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *OS_VERSION , "-sagemaker"]
+    docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /Dockerfile.graviton., *DEVICE_TYPE ]
+    context:
```
Comment (saimidu) [inline pytorch/buildspec-graviton.yml:54]: Is this spec missing the `target` input, to tell it that the build target is `sagemaker`?

File: pytorch/buildspec-graviton.yml [hunk 4]
```diff
@@ -31,9 +34,22 @@ images:
...
+      <<: *INFERENCE_CONTEXT
```
Comment (saimidu) [inline pytorch/buildspec-graviton.yml:55]: nit: Please add a blank line at the end of the file.

File: pytorch/inference/docker/1.12/py3/Dockerfile.graviton.cpu [hunk 5]
```diff
@@ -171,3 +185,55 @@ RUN curl https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.12/license.txt -o /
...
+ && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
+ && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
+ && chmod +x /usr/local/bin/testOSSCompliance \
+ && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
+ && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
+ && rm -rf ${HOME_DIR}/oss_compliance*
+
+EXPOSE 8080 8081
+ENTRYPOINT ["python", "/usr/local/bin/dockerd-entrypoint.py"]
+CMD ["torchserve", "--start", "--ts-config", "/home/model-server/config.properties", "--model-store", "/home/model-server/"]
```
Comment (saimidu) [inline pytorch/inference/docker/1.12/py3/Dockerfile.graviton.cpu:239]: Please add an empty line at the end of the file.

File: pytorch/buildspec-graviton.yml (hunk 1 above)
Comment (xncqr) [reply pytorch/buildspec-graviton.yml]: removed

File: pytorch/buildspec-graviton.yml (hunk 2 above)
Comment (xncqr) [reply pytorch/buildspec-graviton.yml]: added

File: pytorch/buildspec-graviton.yml (hunk 3 above)
Comment (xncqr) [reply pytorch/buildspec-graviton.yml]: Added

File: pytorch/buildspec-graviton.yml (hunk 4 above)
Comment (xncqr) [reply pytorch/buildspec-graviton.yml]: newline added

File: pytorch/inference/docker/1.12/py3/Dockerfile.graviton.cpu (hunk 5 above)
Comment (xncqr) [reply pytorch/inference/docker/1.12/py3/Dockerfile.graviton.cpu]: line added

**Review by saimidu — APPROVED** [review]
